docs/DafnyRef/Statements.md

```diff
@@ -1033,12 +1068,12 @@ If the guard is an asterisk then a non-deterministic choice is made:
   }
 ```
 
-The then alternative of the if-statement must be block statement;
+The then alternative of the if-statement must be a block statement;
 the else alternative may be either a block statement or another if statement.
 The condition of the if statement need not (but may) be enclosed in parentheses.
 
-An if-statement with a binding guard is a ghost statement.
-An if statement with `*` for a guard is non-deterministic.
+An if statement with a binding guard is non-deterministic;
```

Comment:
```suggestion
An if statement with a binding guard is non-deterministic (unless you can prove the unicity separatedly);
```

Reply:
Even if uniqueness can be proved, --enforce-determinism will not allow the construct to be compiled.